backend/api/contacts/src/test/java/co/airy/core/contacts/ImportContactsTest.java

```diff
@@ -0,0 +1,151 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = AirySpringBootApplication.class)
+@TestPropertySource(value = "classpath:test.properties")
+@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+public class ImportContactsTest {
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private WebTestHelper webTestHelper;
+
+    @Autowired
+    private Stores stores;
+
+    @RegisterExtension
+    public static final SharedKafkaTestResource sharedKafkaTestResource = new SharedKafkaTestResource();
+
+    private static KafkaTestHelper kafkaTestHelper;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        kafkaTestHelper = new KafkaTestHelper(sharedKafkaTestResource, Topics.getTopics());
+        kafkaTestHelper.beforeAll();
+    }
+
+    @AfterAll
+    static void afterAll() throws Exception {
+        kafkaTestHelper.afterAll();
+    }
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        webTestHelper.waitUntilHealthy();
+    }
+
+    @Test
+    void canImportContacts() throws Exception {
+        final List<CreateContactPayload> payload = mockContactsListPayload();
+
+        final String content = webTestHelper.post(
+                "/contacts.import",
+                objectMapper.writeValueAsString(payload))
+            .andExpect(status().isCreated()).andReturn().getResponse().getContentAsString();
+
+        List<ContactResponsePayload> contactsResp = objectMapper.readValue(
+                content,
+                new TypeReference<List<ContactResponsePayload>>() {});
+        contactsResp.stream().forEach((contactResponse) -> {
+            try {
+                retryOnException(() -> {
+                    final Contact c = stores.getContact(contactResponse.getId());
+                    assertNotNull(c, "Contact with id %s not found".format(contactResponse.getId()));
+
+                    assertThat(contactResponse.getDisplayName(), equalTo(c.getDisplayName()));
+                    assertThat(contactResponse.getAvatarUrl(), equalTo(c.getAvatarUrl()));
+                    assertThat(contactResponse.getTitle(), equalTo(c.getTitle()));
+                    assertThat(contactResponse.getGender(), equalTo(c.getGender()));
+                }, "The cache is not sync yet");
+            } catch (Exception e) {
+                assertThat(String.format("exception after retryOnException %s", e.getMessage()), false);
+            }
+        });
+    }
+
+    private List<CreateContactPayload> mockContactsListPayload() {
+        return Arrays.asList(
+                new CreateContactPayload(
```

Comment:
This can be more nicely expressed with the `CreateContactPayload.builder()` api.